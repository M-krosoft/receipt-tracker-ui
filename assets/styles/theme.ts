import { createTheme } from "@shopify/restyle";

const palette = {
  purpleLight: "#8C6FF7",
  purplePrimary: "#5A31F4",
  purpleDark: "#3F22AB",

  greenLight: "#56DCBA",
  greenPrimary: "#0ECD9D",
  greenDark: "#0A906E",

  black: "#0B0B0B",
  white: "#F0F2F3",
  red: "#EB5757",
};

const theme = createTheme({
  colors: {
    mainBackground: palette.white,
    cardPrimaryBackground: palette.purplePrimary,
    errorMessage: palette.red,
    successMessage: palette.greenPrimary,

    buttonPrimary: palette.purplePrimary,
    buttonPrimaryForeground: palette.white,

    buttonSecondary: palette.white,
    buttonSecondaryForeground: palette.purplePrimary,
  },
  spacing: {
    auto: "auto",
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
  },
  borderRadii: {
    xs: 2,
    s: 4,
    m: 8,
    l: 16,
    xl: 24,
  },

  textVariants: {
    header1: {
      fontWeight: "bold",
      fontSize: 34,
    },
    header2: {
      fontWeight: "bold",
      fontSize: 28,
    },
    header3: {
      fontWeight: "bold",
      fontSize: 22,
    },
    body1: {
      fontSize: 16,
    },
    body2: {
      fontSize: 14,
    },
    body3: {
      fontSize: 12,
    },
    buttonPrimary: {
      borderWidth: 1,
      borderColor: "buttonPrimary",
      borderStyle: "solid",
      fontWeight: "bold",
      fontSize: 16,
      borderRadius: "m",
      color: "buttonPrimary",
      padding: "s",
    },
    buttonSecondary: {
      fontWeight: "bold",
      fontSize: 16,
      color: "buttonSecondary",
      padding: "s",
      borderRadius: "m",
    },
    defaults: {},
  },

  cardVariants: {
    screenContainer: {
      padding: "l",
    },
    section: {
      padding: "m",
    },
    card: {
      padding: "s",
    },

    defaults: {},
  },

  buttonVariants: {
    buttonPrimary: {
      borderRadius: "m",
      backgroundColor: "buttonPrimaryForeground",
    },

    buttonSecondary: {
      borderRadius: "m",
      backgroundColor: "buttonSecondaryForeground",
    },
    defaults: {},
  },
});

export type Theme = typeof theme;
export default theme;
