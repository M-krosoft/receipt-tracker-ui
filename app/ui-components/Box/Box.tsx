import { PropsWithChildren } from "react";
import { Theme } from "../../../assets/styles/theme";
import {
  BoxProps,
  createRestyleComponent,
  createVariant,
  VariantProps,
} from "@shopify/restyle";

type Props = PropsWithChildren<
  BoxProps<Theme> & VariantProps<Theme, "cardVariants">
>;

export const Box = createRestyleComponent<Props, Theme>([
  createVariant({ themeKey: "cardVariants" }),
]);
