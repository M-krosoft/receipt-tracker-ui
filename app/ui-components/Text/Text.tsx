import { createText } from "@shopify/restyle";
import { Theme } from "../../../assets/styles/theme";

export const Text = createText<Theme>();
