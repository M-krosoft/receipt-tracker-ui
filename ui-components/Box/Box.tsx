import { PropsWithChildren } from "react";
import {
  BoxProps,
  createRestyleComponent,
  createVariant,
  spacing,
  VariantProps,
} from "@shopify/restyle";
import { Theme } from "@assets/styles/theme";

type Props = PropsWithChildren<
  BoxProps<Theme> & VariantProps<Theme, "cardVariants">
>;

export const Box = createRestyleComponent<Props, Theme>([
  spacing,
  createVariant({ themeKey: "cardVariants" }),
]);
