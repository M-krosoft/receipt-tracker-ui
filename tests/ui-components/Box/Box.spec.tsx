import theme from "@assets/styles/theme";
import { ThemeProvider } from "@shopify/restyle";
import { render } from "@testing-library/react-native";
import { Box } from "@ui-components/Box/Box";
import { Text } from "@ui-components/Text/Text";

describe("Box", () => {
  test("should render box with correct styles", async () => {
    // When
    const renderedComponent = await render(
      <ThemeProvider theme={theme}>
        <Box variant="card" margin="m">
          <Text variant="header1">Login</Text>
        </Box>
      </ThemeProvider>
    );

    // Then
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
});
