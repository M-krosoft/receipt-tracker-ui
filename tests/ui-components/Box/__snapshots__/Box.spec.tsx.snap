// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render box with correct styles 1`] = `
<View
  margin="m"
  style={
    [
      {
        "padding": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 34,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Login
  </Text>
</View>
`;
